### PR TITLE
Remove discriminator from ActiveConfiguration and split activeconfiguraiton APIs to config level

### PR DESCRIPTION
**Pull request by yanfa317**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by lirenhe** [comment]
@yanfa317, as this PR contains breaking changes, please follow the process in http://aka.ms/bcforapi to get it reviewed. thanks

**Comment by majastrz** [comment]
> @yanfa317 Fan Yang (AZURE) FTE, as this PR contains breaking changes, please follow the process in http://aka.ms/bcforapi to get it reviewed. thanks

While the changes seem fine from ARM perspective, you need to follow the breaking changes process mentioned by the openapi bot and by @lirenhe.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Review by lirenhe — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveConnectivityConfigurationList.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+  "title": "Get Azure Virtual Network Manager Active Configuration",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveConnectivityConfigurationList.json:2]: >Active Configuration [](http://example.com/codeflow?start=46&length=20)

-> Active Connectivity Configurations?

Comment (kgflying) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveConnectivityConfigurationList.json:2]: The file name should NetworkManagerActiveConfigurationsList.json (plura form)?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{
+  "title": "Get Azure Virtual Network Manager Active Configuration",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:2]: >Configuration [](http://example.com/codeflow?start=53&length=13)

Security Admin Rules?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json [hunk 3]
```diff
@@ -0,0 +1,81 @@
+{
+  "title": "Get Azure Virtual Network Manager Active Configuration",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:2]: >Configuration [](http://example.com/codeflow?start=53&length=13)

.

**Comment by kgflying** [comment]
>       "commitType": "AdminSecurity"

sec admin

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:15 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = False)

**Comment by kgflying** [comment]
>         "/subscriptions/subscriptionC/resourceGroups/resoureGroupSample/providers/Microsoft.Network/networkManagers/testNetworkManager/securityConfigurations/SampleSecurityConfig"

SampleSecurityAdminConfig

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:13 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = False)

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
         "configurationIds": [
           "/subscriptions/subscriptionC/resourceGroups/resoureGroupSample/providers/Microsoft.Network/networkManagers/testNetworkManager/securityConfigurations/SampleSecurityConfig"
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:26]: >SampleSecurityConfig [](http://example.com/codeflow?start=160&length=20)

.

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "region": "westus",
+            "commitTime": "2020-07-10T18:03:22.2578238+05:30",
+            "id": "/subscriptions/subscriptionA/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkManagers/testNetworkManager/securityAdminConfigurations/myTestSecurityConfig/ruleCollections/testRuleCollection/rules/SampleAdminRule",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:22]: >myTestSecurityConfig [](http://example.com/codeflow?start=170&length=20)

myTestSecurityAdminConfig should be better?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "region": "westus",
+            "commitTime": "2020-07-10T18:03:22.2578238+05:30",
+            "id": "/subscriptions/subscriptionA/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkManagers/testNetworkManager/securityUserConfigurations/myTestSecurityConfig/ruleCollections/testRuleCollection/rules/SampleUserRule",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:22]: >myTestSecurityConfig [](http://example.com/codeflow?start=169&length=20)

myTestSecurityUserConfig

Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:22]: >SampleUserRule [](http://example.com/codeflow?start=231&length=14)

SampleSecurityUserRule?

**Comment by kgflying** [comment]
>         "/subscriptions/subscriptionC/resourceGroups/resoureGroupSample/providers/Microsoft.Network/networkManagers/testNetworkManager/securityConfigurations/SampleSecurityConfig"

securityAdminConfigurations

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:13 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = True)

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json (hunk 4 above)
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:26]: >securityConfigurations [](http://example.com/codeflow?start=137&length=22)

.

**Comment by kgflying** [comment]
>   "title": "Post Azure Virtual Network Manager Commit",

Do we need a sample for connectivity and sec user?

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerCommitPost.json:2 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = True)

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    "parameters": {
+      "$skipToken": "FakeSkipTokenCode"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/subscriptionA/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkManagers/testNetworkManager/securityAdminConfigurations/myTestSecurityConfig/ruleCollections/testRuleCollection/rules/SampleAdminRule",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json:17]: >SampleAdminRule [](http://example.com/codeflow?start=232&length=15)

.

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+            "configurationDisplayName": "SampleConfig",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:23]: >SampleConfig [](http://example.com/codeflow?start=41&length=12)

I suggest change to SampleSecurityAdminConfig.

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            "configurationDescription": "SampleDescription",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:24]: >SampleDescription [](http://example.com/codeflow?start=41&length=17)

I suggest change to SampleSecurityAdminDescription

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+            "configurationDisplayName": "SampleConfig",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:23]: >SampleConfig [](http://example.com/codeflow?start=41&length=12)

I suggest change to SampleSecurityUserConfig.

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+            "configurationDescription": "SampleDescription",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:24]: >SampleDescription [](http://example.com/codeflow?start=41&length=17)

I suggest change to SampleSecurityUserDescription

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+            "configurationDisplayName": "SampleConfig",
+            "configurationDescription": "SampleDescription",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json:19]: .

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json (hunk 7 above)
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityAdminRulesList.json:17]: >myTestSecurityConfig [](http://example.com/codeflow?start=170&length=20)

myTestSecurityAdminConfig

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityUserRulesListBySubnet.json [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    "parameters": {
+      "$skipToken": "FakeSkipTokenCode"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/subscriptionA/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkManagers/testNetworkManager/securityUserConfigurations/myTestSecurityConfig/ruleCollections/testRuleCollection/rules/SampleUserRule",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityUserRulesListBySubnet.json:18]: >myTestSecurityConfig [](http://example.com/codeflow?start=169&length=20)

myTestSecurityAdminConfig

Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityUserRulesListBySubnet.json:18]: >SampleUserRule [](http://example.com/codeflow?start=231&length=14)

SamleSecurityUserRule

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityUserRulesListBySubnet.json [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+            "configurationDisplayName": "SampleConfig",
+            "configurationDescription": "SampleDescription",
```
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerEffectiveSecurityUserRulesListBySubnet.json:20]: .

**Comment by kgflying** [comment]
>             "etag": "sadf-asdf-asdf-asdf",

This is not valid. I think we should list a valid guid here

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerListAll.json:15 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = True)

**Comment by kgflying** [comment]
>         "etag": "sadf-asdf-asdf-asdf",

.

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPatch.json:21 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = True)

**Comment by kgflying** [comment]
>   "title": "Patch Managed Network Tags",

Is the NM update?
We don't support NM, do we?

If not, we should remove this file.

---
Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPatch.json:2 in b8b5ffb. [](commit_id = b8b5ffb98f52a89186871a0424a00ed0be904b00, deletion_comment = True)

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json (hunk 5 above)
Comment (kgflying) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:22]: >SampleAdminRule [](http://example.com/codeflow?start=232&length=15)

SampleSecurityAdminRule?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json (hunk 6 above)
Comment (yanfa317) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityUserRulesList.json:22]: It is just an example name, any string would work

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json (hunk 8 above)
Comment (yanfa317) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerActiveSecurityAdminRulesList.json:23]: Any string works here

**Comment by yanfa317** [comment]
> > "title": "Patch Managed Network Tags",
> 
> Is the NM update?
> We don't support NM, do we?
> 
> If not, we should remove this file.
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPatch.json:2 in [b8b5ffb](https://github.com/Azure/azure-rest-api-specs/commit/b8b5ffb98f52a89186871a0424a00ed0be904b00). [](commit_id = [b8b5ffb](https://github.com/Azure/azure-rest-api-specs/commit/b8b5ffb98f52a89186871a0424a00ed0be904b00), deletion_comment = True

It is required APi, and we need to support it

**Comment by kgflying** [comment]
@lirenhe , we still before public preview, do we need follow [http://aka.ms/bcforapi](http://aka.ms/bcforapi) ? Just wondering

---
In reply to: [884652089](https://github.com/Azure/azure-rest-api-specs/pull/15282#issuecomment-884652089) [](http://example.com/codeflow?ancestors=884652089,884652089)

**Comment by chiragg4u** [comment]
Looks like All your list calls are modeled as POST! 
Whereas for most control plane features as well as automated clients GET is what being used see guidance @ Azure Policy Guidelines | ARM Wiki
POST is applicable only when we would like to return secrets which needs to be hidden in GET by default. 

Please confirm the reasoning for this design apart from following existing pattern (which may have issues and not to be continued)

**Comment by raosuhas** [comment]
Changes look ok. But please resolve breaking changes or get exception. See changelog above for details.

**Comment by yanfa317** [comment]
We have talked with @chiragg4u by mail, and have resolved his concerns. I think he has approved it already. thanks!

**Pull request merged**